### PR TITLE
feat: add animation for add favorites

### LINKED ISSUE #4305: Favorited icon animation
### Description

Please add a new animation for favorite feature. (the demo is attached in the attachment file, if you have any question please let me know)

https://github.com/toeverything/AFFiNE/assets/40333372/81698f13-27f3-46b4-a0ab-310db04f15f3

[Favorite_animation.zip](https://github.com/toeverything/AFFiNE/files/12573972/Favorite_animation.zip)

### Use case

Use case: When user clicks favorite icon the animation will run **once**. 

**Used to add page/edgeless to favorited**

Special case: If the user adds it to the favorited list through the details list, there's no need to display the animation.

image

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by JimmFly**
Close #4305 

https://github.com/toeverything/AFFiNE/assets/102217452/427648b6-4966-4704-a33d-7445c06e18e6

### CODE REVIEW

**Review by zanwei — APPROVED** [review]

File: packages/component/src/components/page-list/components/favorite-tag.tsx [hunk 1]
```diff
@@ -5,28 +5,48 @@ import {
...
 
 export const FavoriteTag = forwardRef<
   HTMLButtonElement,
   {
     active: boolean;
   } & Omit<IconButtonProps, 'children'>
 >(({ active, onClick, ...props }, ref) => {
+  const [playAnimation, setPlayAnimation] = useState(false);
   const t = useAFFiNEI18N();
+  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
```
Comment (joooye34) [inline packages/component/src/components/page-list/components/favorite-tag.tsx:21]: why not to use useCallback hook, and add active to be a dependency, then there just need one handleClick

File: packages/component/src/components/page-list/components/favorite-tag.tsx [hunk 2]
```diff
@@ -5,28 +5,48 @@ import {
...
+    e.stopPropagation();
+    onClick?.(e);
+    setPlayAnimation(true);
+  };
+  const handleFavoritedClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.stopPropagation();
+    onClick?.(e);
+    setPlayAnimation(false);
+  };
```
Comment (pengx17) [inline packages/component/src/components/page-list/components/favorite-tag.tsx:30]: useCallback

File: packages/component/src/components/page-list/components/favorite-tag.tsx [hunk 3]
```diff
@@ -5,28 +5,54 @@ import {
...
+    [onClick]
+  );
+  const handleFavoritedClick = useCallback(
+    (e: React.MouseEvent<HTMLButtonElement>) => {
+      e.stopPropagation();
+      onClick?.(e);
+      setPlayAnimation(false);
+    },
+    [onClick]
+  );
```
Comment (lawvs) [inline packages/component/src/components/page-list/components/favorite-tag.tsx:36]:
```suggestion
  const handleClick = useCallback(
    (e: React.MouseEvent<HTMLButtonElement>) => {
      e.stopPropagation();
      onClick?.(e);
      setPlayAnimation(active);
    },
    [onClick]
  );
```

File: packages/component/src/components/page-list/components/favorite-tag.tsx [hunk 4]
```diff
@@ -5,28 +5,54 @@ import {
...
   return (
     <Tooltip content={active ? t['Favorited']() : t['Favorite']()} side="top">
       <IconButton
         ref={ref}
         active={active}
-        onClick={e => {
-          e.stopPropagation();
-          onClick?.(e);
-        }}
+        onClick={active ? handleFavoritedClick : handleClick}
```
Comment (lawvs) [inline packages/component/src/components/page-list/components/favorite-tag.tsx:42]:
```suggestion
        onClick={handleClick}
```

**Review by pengx17 — APPROVED** [review]

**Pull request merged**